### PR TITLE
SubGhz: reading keys from encrypted files

### PR DESCRIPTION
**Pull request by Skorpionm**
# What's new

- SubGhz: add file with manufactory codes, and the ability to add your own manufactory codes for KeeLog
- SubGhz: add encrypt RAW data, add decrypt and get RAW data
- SubGhz: add encrypt magic_xor_atomo, parsing atomo using file encrypt
- SubGhz: add file encrypt nice_flor_s_tx and nice_flor_s_rx, parsing Nice FLOR S using file encrypt
- SubGhz: fix name load file

# Verification 

- Compile load via qFlipper
- Download SubGhz app -> Read, check reception Keeloq, StarLine, CAME Atomo, Nice FLOR-S

# Checklist (do not modify)

- [x] PR has description of feature/bug or link to Confluence/Jira task
- [X] Description contains actions to verify feature/bugfix
- [X] I've built this code, uploaded it to the device and verified feature/bugfix

### CODE REVIEW

File: assets/resources/subghz/keeloq_mfcodes_user [hunk 1]
```diff
@@ -0,0 +1,11 @@
+# for adding manufacture keys
+# AABBCCDDEEFFAABB:X:NAME\r\n
+# AABBCCDDEEFFAABB - man 64 bit
+# X - encryption method 1 - Simple Learning, 2 - Normal_Learning, 
+#   	0 - iterates over both previous and man in direct and reverse byte sequence
+# NAME - name (string without spaces) max 64 characters long
+Filetype: Flipper SubGhz Keystore File
+Version: 0
+Encryption: 0
+AABBCCDDEEFFAABB:1:Test1
```
Comment (skotopes) [inline assets/resources/subghz/keeloq_mfcodes_user:10]: Those will be loaded. Probably make sense to comment them out.

File: lib/subghz/subghz_keystore.c [hunk 2]
```diff
@@ -315,3 +325,268 @@ SubGhzKeyArray_t* subghz_keystore_get_data(SubGhzKeystore* instance) {
...
+    flipper_file_free(input_flipper_file);
+
+    free(encrypted_line);
+
+    furi_record_close("storage");
+
+    return encrypted;
+}
+
+bool subghz_keystore_raw_get_data(const char* file_name, size_t bias, uint8_t* data, size_t len) {
```
Comment (skotopes) [inline lib/subghz/subghz_keystore.c]: `bias` -> `offset`

File: lib/subghz/subghz_keystore.c [hunk 3]
```diff
@@ -315,3 +325,268 @@ SubGhzKeyArray_t* subghz_keystore_get_data(SubGhzKeystore* instance) {
...
+        char buffer[bufer_size];
+        size_t ret = 0;
+        bool decrypted = true;
+        //skip the end of the previous line "\n"
+        storage_file_read(file, buffer, 1);
+
+        size_t size = storage_file_size(file);
+        size -= storage_file_tell(file);
+        if(size < (bias * 2 + len * 2)) {
+            FURI_LOG_E(SUBGHZ_KEYSTORE_TAG, "Went outside the file");
```
Comment (skotopes) [inline lib/subghz/subghz_keystore.c]: `"Went outside the file"` -> `"Seek position exceeds file size"`

**Review by skotopes — CHANGES_REQUESTED** [review]

File: lib/subghz/subghz_keystore.c (hunk 3 above)
Comment (Skorpionm) [reply lib/subghz/subghz_keystore.c]: ok

File: lib/subghz/subghz_keystore.c (hunk 2 above)
Comment (Skorpionm) [reply lib/subghz/subghz_keystore.c]: ok

File: assets/resources/subghz/keeloq_mfcodes_user (hunk 1 above)
Comment (Skorpionm) [reply assets/resources/subghz/keeloq_mfcodes_user:10]: let it remain so clearly and does not break anything

**Review by skotopes — APPROVED** [review]

**Pull request merged**